decidim-core/app/models/decidim/action_log.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This class represents an action of a user on a resource. It is used
+  # for transparency reasons, to log all actions so all other users can
+  # see the actions being performed.
+  class ActionLog < ApplicationRecord
+    belongs_to :organization,
+               foreign_key: :decidim_organization_id,
+               class_name: "Decidim::Organization"
+
+    belongs_to :user,
+               foreign_key: :decidim_user_id,
+               class_name: "Decidim::User"
+
+    belongs_to :feature,
+               foreign_key: :decidim_feature_id,
+               optional: true,
+               class_name: "Decidim::Feature"
+
+    belongs_to :resource,
+               polymorphic: true
+
+    belongs_to :participatory_space,
+               optional: true,
+               polymorphic: true
+
+    validates :organization, :user, :action, :resource, presence: true
```

Comment:
Should we add something like:

```ruby
def readonly?
  !new_record?
end
```

To make sure no one can alter the logs easily?

Reply:
Oh, I didn't know about that. I'll add it, yes!

Reply:
Done